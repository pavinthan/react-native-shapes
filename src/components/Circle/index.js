import Circle from './Circle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Circle.propTypes = propTypes;
Circle.defaultProps = defaultProps;

export default Circle;
