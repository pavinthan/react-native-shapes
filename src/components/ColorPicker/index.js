import ColorPicker from './ColorPicker';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

ColorPicker.propTypes = propTypes;
ColorPicker.defaultProps = defaultProps;

export default ColorPicker;
