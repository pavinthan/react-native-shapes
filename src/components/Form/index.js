import Form from './Form';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
