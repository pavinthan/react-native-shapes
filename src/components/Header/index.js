import Header from './Header';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
