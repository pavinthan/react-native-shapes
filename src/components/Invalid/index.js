import Invalid from './Invalid';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Invalid.propTypes = propTypes;
Invalid.defaultProps = defaultProps;

export default Invalid;
