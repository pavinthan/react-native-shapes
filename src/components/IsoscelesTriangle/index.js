import IsoscelesTriangle from './IsoscelesTriangle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

IsoscelesTriangle.propTypes = propTypes;
IsoscelesTriangle.defaultProps = defaultProps;

export default IsoscelesTriangle;
