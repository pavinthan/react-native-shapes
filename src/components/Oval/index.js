import Oval from './Oval';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Oval.propTypes = propTypes;
Oval.defaultProps = defaultProps;

export default Oval;
