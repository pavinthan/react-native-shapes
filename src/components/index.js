export { default as Circle } from './Circle';
export { default as ColorPicker } from './ColorPicker';
export { default as Form } from './Form';
export { default as Header } from './Header';
export { default as Invalid } from './Invalid';
export { default as IsoscelesTriangle } from './IsoscelesTriangle';
export { default as Oval } from './Oval';
export { default as Parallelogram } from './Parallelogram';
export { default as Polygon } from './Polygon';
export { default as Rectangle } from './Rectangle';
export { default as Whiteboard } from './Whiteboard';
