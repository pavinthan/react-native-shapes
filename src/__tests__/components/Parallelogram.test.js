import React from 'react';
import { Parallelogram } from '../../components';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer
    .create(<Parallelogram width={100} height={200} fill="#000" />)
    .toJSON();
  expect(rendered).toBeTruthy();
});

it('renders square component', () => {
  const tree = renderer
    .create(<Parallelogram width={100} height={200} fill="#000" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
