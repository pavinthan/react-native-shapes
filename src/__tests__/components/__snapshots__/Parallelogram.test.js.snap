// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders square component 1`] = `
<ARTSurfaceView
  style={
    Array [
      undefined,
      Object {
        "height": 200,
        "width": 150,
      },
    ]
  }
>
  <ARTGroup
    opacity={1}
    transform={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
  >
    <ARTShape
      d={
        Array [
          0,
          50,
          0,
          2,
          150,
          0,
          2,
          100,
          200,
          2,
          0,
          200,
          2,
          50,
          0,
          1,
        ]
      }
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      opacity={1}
      stroke={null}
      strokeCap={1}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
  </ARTGroup>
</ARTSurfaceView>
`;
